### PR TITLE
`save_handler` auto detects `DiskSaver` when path passed

### LINKED ISSUE #2194: Automatically detect `DiskSaver` as `save_handler` in `Checkpoint`
## 🚀 Feature

Currently, when we use `Checkpoint`, we have to specify a `save_handler` even when we want to store the checkpoints on disk:
```python
handler = Checkpoint(
        to_save, DiskSaver('/tmp/models', create_dir=True), n_saved=2
   )
```
An enhancement can be to automatically detect the `save_handler` as `DiskSaver` if a path is passed, like this:
```python
handler = Checkpoint(
        to_save, create_dir=True, dir_name='/tmp/models', n_saved=2
   )
```
Internally, 
```python
if dir_name is not None:
    save_handler = DiskSaver(path, create_dir = create_dir)
```

### PR DESCRIPTION
**Pull request by Priyansi**
Fixes #2194 

Description: Added additional parameter to `Checkpoint`'s `save_handler` to accept a string containing the `dirname`.

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by Priyansi** [comment]
@vfdev-5 should I also add an example in the doc string and/or a test?

File: ignite/handlers/checkpoint.py [hunk 1]
```diff
@@ -286,15 +287,17 @@ def __init__(
...
+            raise TypeError("Argument `save_handler` should be a string or callable or inherit from BaseSaveHandler")
 
         if global_step_transform is not None and not callable(global_step_transform):
             raise TypeError(f"global_step_transform should be a function, got {type(global_step_transform)} instead.")
 
         self.to_save = to_save
         self.filename_prefix = filename_prefix
         self.save_handler = save_handler
+        if isinstance(self.save_handler, str):
+            self.save_handler = DiskSaver(self.save_handler, create_dir=True)
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:300]: mypy is not happy with this type redefinition. Let's recode that as 
```python
if isinstance(save_handler, str):
    self.save_handler = DiskSaver(save_handler, create_dir=True)
else:
    self.save_handler = save_handler
```
Please check locally if this helps to solve mypy issue

**Review by vfdev-5 — COMMENTED** [review]
@Priyansi  thanks for the PR. Yes, please add a test. For the docstring, we can just update all example and remove DiscSaver usage and add an example showing its usage. Such that we show the simplified API and gives one example of how to do with DiskSaver.
Can you also update all other examples by removing DiskSaver if used with Checkpoint ?

**Comment by vfdev-5** [comment]
@Priyansi can we finalize this PR this week

**Comment by Priyansi** [comment]
Hey @vfdev-5 , I apologize for the delay. Could you please re-review my PR?

**Comment by sdesrozis** [comment]
@Priyansi it seems some tests are failing...

**Comment by Priyansi** [comment]
Hey @sdesrozis , the test that is failing:
```
def test_kwargs_loss():
    loss = Loss(nll_loss)

    y_pred, y, _ = y_test_1()
    loss.update((y_pred, y, {"weight": torch.tensor([0, 0, 0], dtype=torch.float)}))
    assert_almost_equal(loss.compute(), 0)
```
isn't related to any changes that I made in `Checkpoint`.

**Comment by vfdev-5** [comment]
> Hey @sdesrozis , the test that is failing:
> 
> ```
> def test_kwargs_loss():
>     loss = Loss(nll_loss)
> 
>     y_pred, y, _ = y_test_1()
>     loss.update((y_pred, y, {"weight": torch.tensor([0, 0, 0], dtype=torch.float)}))
>     assert_almost_equal(loss.compute(), 0)
> ```
> 
> isn't related to any changes that I made in `Checkpoint`.

See https://github.com/pytorch/ignite/issues/2240

**Comment by Priyansi** [comment]
> @Priyansi see #2240

@vfdev-5 Got it! How should I proceed with this?

The other broken test has been fixed.

**Comment by vfdev-5** [comment]
@Priyansi i'll fix it here: https://github.com/pytorch/ignite/pull/2242

File: ignite/handlers/checkpoint.py [hunk 2]
```diff
@@ -245,6 +245,14 @@ class Checkpoint(Serializable):
             trainer.run(data_loader, max_epochs=10)
             > ["best_model_9_accuracy=0.77.pt", "best_model_10_accuracy=0.78.pt", ]
 
+        Customize the `save_handler`:
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:248]:
```suggestion
        Customize the ``save_handler``:
```

**Comment by Priyansi** [comment]
@vfdev-5 thank you for your suggestions. I have updated the docstrings! Could you please re-review this?

File: ignite/handlers/checkpoint.py [hunk 3]
```diff
@@ -244,6 +245,14 @@ class Checkpoint(Serializable):
...
 
+        Customise the ``save_handler``:
+
+        .. code-block:: python
+
+            handler = Checkpoint(
+                to_save, save_handler=DiskSaver('/tmp/models', create_dir=True, **kwargs), n_saved=2
+            )
+
     .. versionchanged:: 0.4.3
```
Comment (vfdev-5) [inline ignite/handlers/checkpoint.py:256]: Also we forget about `.. versionchanged:: 0.5.0` for this change

Comment (Priyansi) [reply ignite/handlers/checkpoint.py:256]: I hope the note in the current commit is okay.

Comment (vfdev-5) [reply ignite/handlers/checkpoint.py:256]: Looks good

**Review by vfdev-5 — APPROVED** [review]
Thanks for the PR @Priyansi

**Pull request merged**